src/Adapter/Translations/TranslationRouteFinder.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Translations;
+
+use Link;
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleRepositoryInterface;
+use PrestaShopBundle\Service\TranslationService;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+
+/**
+ * Class TranslationRouteFinder finds the correct route for translations
+ */
+class TranslationRouteFinder
+{
+    /**
+     * @var TranslationService
+     */
+    private $translationService;
+    /**
+     * @var Link
+     */
+    private $link;
+    /**
+     * @var ModuleRepositoryInterface
+     */
+    private $moduleRepository;
+
+    /**
+     * @param TranslationService $translationService
+     * @param Link $link
+     * @param ModuleRepositoryInterface $moduleRepository
+     */
+    public function __construct(
+        TranslationService $translationService,
+        Link $link,
+        ModuleRepositoryInterface $moduleRepository
+    ) {
+        $this->translationService = $translationService;
+        $this->link = $link;
+        $this->moduleRepository = $moduleRepository;
+    }
+
+    /**
+     * Finds the correct translation route out of given query
+     *
+     * @param ParameterBag $query
+     *
+     * @return string
+     */
+    public function findRoute(ParameterBag $query)
+    {
+        $routeProperties = $query->get('modify_translations');
+        $propertyAccessor = PropertyAccess::createPropertyAccessor();
+        $route = 'admin_international_translation_overview';
+
+        switch ($propertyAccessor->getValue($routeProperties, '[translation_type]')) {
+            case 'mails':
```

Comment:
shouldnt these case params be defined as constants as they are used multiple times?